### PR TITLE
[Core] Improve config_list_from_json

### PR DESCRIPTION
**Pull request by BeibinLi**
#### UPDATE
The `env_or_file` variabe can point to an environment variable of file path now.
In this new update, we check if the variable is a file or a json string. If it is a file, then it will load from the file.

## Why are these changes needed?
It is a little bit confusing to the previous `config_list_from_json` function, where the environment variable would store a JSON string. It is usually more intuitive such that an environment variable points to a file location. 

## Checks

- [] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [X] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [X] I've made sure all auto checks have passed.

### CODE REVIEW

**Review by ekzhu — CHANGES_REQUESTED** [review]
This is helpful. It also addresses the issue with VScode's Jupyter notebook environment persisting old environment variables. 

Updating this example notebook too? https://github.com/microsoft/autogen/blob/main/notebook/oai_openai_utils.ipynb

**Review by ekzhu — APPROVED** [review]

**Comment by ekzhu** [comment]
I think this change is fine. I anticipate larger change coming regarding the configuration utilities due to https://github.com/microsoft/autogen/issues/1095. So we don't need to update the notebook about this small feature. Merging this PR right now give us the functionality target we need for the rewrite.

**Comment by ekzhu** [comment]
@sonichi

**Comment by sonichi** [comment]
There are conflicts to resolve.

**Comment by ekzhu** [comment]
@BeibinLi do you want to take a look at the conflicts. I tried to manually merge it but lost the context when I am trying to edit the test case.

**Comment by BeibinLi** [comment]
@sonichi  @ekzhu  Thanks for the suggestions! I have merged the new changes.

We can hold the notebook update until the Pydantic PR.

File: autogen/oai/openai_utils.py [hunk 1]
```diff
@@ -404,7 +405,8 @@ def config_list_from_json(
     of acceptable values for that field, the configuration will still be considered a match.
 
     Args:
-        env_or_file (str): The name of the environment variable or the filename containing the JSON data.
+        env_or_file (str):The name of the environment variable, the filename, or the environment variable of the filename
```
Comment (sonichi) [inline autogen/oai/openai_utils.py:408]:
```suggestion
        env_or_file (str): The name of the environment variable, the filename, or the environment variable of the filename
```

File: autogen/oai/openai_utils.py [hunk 2]
```diff
@@ -423,10 +425,20 @@ def config_list_from_json(
...
     """
-    json_str = os.environ.get(env_or_file)
-    if json_str:
+    env_str = os.environ.get(env_or_file)
+
+    if env_str:
+        # The environment variable exists. We should use information from it.
+        if os.path.exists(env_str):
+            # It is a file location, and we need to load the json from the file.
+            json_str = open(env_str).read()
```
Comment (sonichi) [inline autogen/oai/openai_utils.py:434]: shall we use "with" for proper file operations?

Comment (BeibinLi) [reply autogen/oai/openai_utils.py]: Good suggestion. Resolved.

File: test/oai/test_utils.py [hunk 3]
```diff
@@ -1,10 +1,13 @@
-import os
 import json
-import pytest
 import logging
+import os
+import sys
```
Comment (sonichi) [inline test/oai/test_utils.py:4]: is this import unused?

File: notebook/coding/bubble_sort.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# filename: bubble_sort.py
```
Comment (ekzhu) [inline notebook/coding/bubble_sort.py:1]: Is this file accidentally checked in?

**Pull request merged**